gbols/api-endpoint
# #160158014 Bg encryption wildcards

## What does this PR do?
- validates user input name and makes sure there are no wildcards in it.
- Encrypts user password as an added security layer.

## Description of Tasks to be completed
- Write tests for this functionality

### How should this be manually tested 
- pull from this branch `bg-encryption-wildcards-160158014` into a new branch.
- run npm install
- Endpoint should look like this `/auth/signup.html` relative to your environment.

### What are the relevant pivotal tracker stories.

- #160158014 username should be restricted to only alphabets and password encryption.
![screenshot from 2018-08-30 02-04-18](https://user-images.githubusercontent.com/27724644/44824961-237a5f80-ac00-11e8-86ad-88e370258242.png)

![screenshot from 2018-08-30 02-03-57](https://user-images.githubusercontent.com/27724644/44824964-29704080-ac00-11e8-8521-d4b18d27bb70.png)
